### DISCUSSION TITLE
How to make type hinting work for named function handler with validator/zValidator

### DISCUSSION THREAD

**Discussion by gouku** [honojs/hono#2511] [Q&A]
I have a route:

```ts
app.post("/add", zValidator("json", schema), async (c) => {
  const body = c.req.valid("json")
  ...
}
```

This works and has the correct typing but:

```ts
app.post("/add", zValidator("json", schema), AddHandler)

// handlers/add.ts
export const AddHandler = async (c: Context) => {
  const body = c.req.valid("json")
  ...
}
```

does not work:

<img width="602" alt="image" src="https://github.com/honojs/hono/assets/9361308/21ed709a-6698-4c15-aefd-841bc700c4f8">

Any help would be appreciated.

**Comment by NicoPlyley** [comment]
That's because addHandler has no types associated with it. The .post function provides types to it. It is recommended to keep handler all together, a lot of people got used to ruby on rails splitting them up.

However, if you do want to split them there is an experimental way to do that right now which you can see the documentation about [here](https://hono.dev/helpers/factory#factory-createhandlers)

**Reply by gouku** [reply]
Code splitting is important for me as the codebase is growing we can not put all the handlers in one file.

I don't know how to use the createHandlers because I'm not a senior developer. I'll look into it. Thanks for the info!

**Reply by NicoPlyley** [reply]
Go ahead and try it out, and if you have any questions let me know

**Reply by tconroy** [reply]
It looks like the docs don't really provide much examples -- for people new to the framework, it's not intuitive. Using `factory.createHandlers`, how do you define multiple handlers (i.e GET, POST, etc) on one factory instance?
